### PR TITLE
feat: rework modals

### PR DESCRIPTION
**Pull request by dzbo**
### Ticket ID

[DEV-11664](https://app.clickup.com/t/2645698/DEV-11664)
[DEV-11717](https://app.clickup.com/t/2645698/DEV-11717)

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-11664 Rework all modals to use query params](https://app.clickup.com/t/2645698/DEV-11664)

**Review by richtera — APPROVED** [review]

File: composables/useModal.ts [hunk 1]
```diff
@@ -1,49 +1,46 @@
...
+    modalSize: modal.size || 'small',
+    modalData: JSON.stringify(modal.data),
   }
 
-  if (typeof modal === 'string') {
-    appStore.setModal({ ...modalDefaults, message: modal })
-  } else {
-    appStore.setModal({ ...modalDefaults, ...modal })
-  }
+  await navigateTo({
```
Comment (doubleppereira) [inline composables/useModal.ts:22]: Not sure I understand the idea of navigate here. So on opening/closing modal you navigate?

File: components/ModalTemplateSwitchExtensionNetwork.vue [hunk 2]
```diff
@@ -17,7 +12,7 @@ const handleChangeNetwork = async () => {
   } catch (error: unknown) {
     console.error(error)
   } finally {
-    props.closeModal()
+    closeModal()
```
Comment (doubleppereira) [inline components/ModalTemplateSwitchExtensionNetwork.vue:15]: wouldnt you need to do `await` here?

File: composables/useModal.ts (hunk 1 above)
Comment (dzbo) [reply composables/useModal.ts:22]: `showModal` / `hideModal` just change query params. Later in `AppModal.vue` there is a watcher that check for params and fill component with right template and data.

File: components/ModalTemplateSwitchExtensionNetwork.vue (hunk 2 above)
Comment (dzbo) [reply components/ModalTemplateSwitchExtensionNetwork.vue]: Seem to be not necessary but wont hurt ;)

File: composables/useModal.ts (hunk 1 above)
Comment (doubleppereira) [reply composables/useModal.ts:22]: Ok I never seen a modal changing route params that's why but ok I understand now

**Review by doubleppereira — APPROVED** [review]

File: composables/useModal.ts (hunk 1 above)
Comment (dzbo) [reply composables/useModal.ts:22]: It's quite innovative solution I agree 😅 Fabs wanted to be able to re-open modals with browser back button.

**Pull request merged**